public/app/core/directives/value_select_dropdown.ts

```diff
@@ -17,29 +24,38 @@ export class ValueSelectDropdownCtrl {
 
   hide: any;
   onUpdated: any;
+  queryHasSearchFilter: boolean;
+  debouncedQueryChanged: Function;
 
   /** @ngInject */
-  constructor(private $q: any) {}
+  constructor(private $q: any, private $scope: IScope) {
+    this.queryHasSearchFilter = this.variable ? containsSearchFilter(this.variable.query) : false;
+    this.debouncedQueryChanged = debounce(this.queryChanged.bind(this), 200);
```

Comment:
Not sure if this an Ok debounce, 100ms might be enough?

Reply:
no, think 200 is a good choice.